test/DefaultCluster.Tests/GrainReferenceTest.cs

```diff
@@ -248,7 +248,7 @@ private T NewtonsoftJsonSerializeRoundtrip<T>(T obj)
             var settings = OrleansJsonSerializer.GetDefaultSerializerSettings(this.HostedCluster.SerializationManager, this.GrainFactory);
             // http://james.newtonking.com/json/help/index.html?topic=html/T_Newtonsoft_Json_JsonConvert.htm
             string json = JsonConvert.SerializeObject(obj, settings);
-            object other = JsonConvert.DeserializeObject(json, settings);
+            object other = JsonConvert.DeserializeObject(json, typeof(T), settings);
```

Comment:
nevermind, this was me looking at just the last commit, but looking at the entire change, you actually got to a similar place as you started (except by being more restrictive when using `typeof(T)` instead of `obj.GetType()`.
Move along